pkg/chunked/storage_linux.go

```diff
@@ -947,11 +967,6 @@ func (c *chunkedDiffer) retrieveMissingFiles(stream ImageSourceSeekable, dirfd i
 		}
 
 		if _, ok := err.(ErrBadRequest); ok {
-			// If the server cannot handle at least 64 chunks in a single request, just give up.
-			if len(chunksToRequest) < 64 {
-				return err
-			}
-
```

Comment:
AFAICS this can now loop infinitely, with a single-element `chunksToRequest`.

Reply:
It should not happen because if we got so far the server at least served the range request for the TOC, but let's be safe (could be a malicious server): I've added a check to fail if `chunksToRequest == 1`.